src/data/nat/prime.lean

```diff
@@ -190,6 +198,15 @@ section min_fac
     ((dvd_prime pp).1 fd).resolve_left (ne_of_gt f2)⟩,
    λ ⟨p2, e⟩, e ▸ min_fac_prime (ne_of_gt p2)⟩
 
+  /--
+  This instance is faster in the virtual machine than `decidable_prime_1`.
+  However it uses `classical.choice`, so can't actually be used in a
+  decision procedure.
```

Comment:
It doesn't use classical.choice, but it does use well founded recursion, which the kernel doesn't unfold very well because it has to unfold proofs. The VM just goes ahead and executes the recursion without worrying about it, which is why it is fast in the VM.

Reply:
These explanations are leaving me confused as to why `dec_trivial` with the default instance actually fails, rather than just runs slowly. :-(